### PR TITLE
OCM-4459 | fix: change casing to fix the create failure issue

### PR DESCRIPTION
**Pull request by chenz4027**
https://issues.redhat.com/browse/OCM-4459

### CODE REVIEW

File: model/clusters_mgmt/v1/external_auth_type.model [hunk 1]
```diff
@@ -84,7 +84,7 @@ struct ExternalAuthClaim {
 struct TokenClaimMappings {
 
     // Username is a name of the claim that should be used to construct usernames for the cluster identity.
-    UserName UsernameClaim
+    Username UsernameClaim
```
Comment (machi1990) [inline model/clusters_mgmt/v1/external_auth_type.model:87]: You could also use the json annotation
```
 @json(name = "username")
 UserName UsernameClaim
```
This will ensure backward compatible for the `UserName` field, assuming that the sdk as already been release.

File: CHANGES.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 This document describes the relevant changes between releases of the API model.
 
+## 0.0.362 Mar 06 2024
+- Changed `UserName` attribute for TokenClaimMappings to `Username`.
+
```
Comment (machi1990) [inline CHANGES.md:7]: These changes are typically done on a separate MR once the main PR to introduce the changes has been merged; https://github.com/openshift-online/ocm-sdk-go/blob/main/CONTRIBUTING.md#releasing-a-new-ocm-api-model-version

File: model/clusters_mgmt/v1/external_auth_type.model (hunk 1 above)
Comment (andreadecorte) [reply model/clusters_mgmt/v1/external_auth_type.model:87]: I am not aware of anyone using this yet so it should be safe. But we can also change the json annotation if we want

Comment (machi1990) [reply model/clusters_mgmt/v1/external_auth_type.model:87]: Yes. And if there were people using it, it's probably not working so we should be fine just changing it now. I wanted to share the above as mostly an FYI

File: CHANGES.md (hunk 2 above)
Comment (chenz4027) [reply CHANGES.md:7]: Thanks @machi1990 , will remove this commit

File: model/clusters_mgmt/v1/external_auth_type.model (hunk 1 above)
Comment (chenz4027) [reply model/clusters_mgmt/v1/external_auth_type.model:87]: Thanks @machi1990 @andreadecorte , let me add the json annotation

**Comment by chenz4027** [comment]
@andreadecorte @machi1990 Would you mind taking another look? Thanks!

**Review by machi1990 — APPROVED** [review]
I'll let @andreadecorte give this another look and final approval as the SME

**Comment by chenz4027** [comment]
Hmm seems like it's not auto merged. @andreadecorte @machi1990 Do you have merge permission by any chance?

**Comment by machi1990** [comment]
@chenz4027 it is merged now, you can followup with the rest of the changes
